includes/wikia/api/RevisionUpvotesApiController.class.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+class RevisionUpvotesApiController extends WikiaApiController {
+
+	/**
+	 * Add upvote for given revision made by current user
+	 *
+	 * @requestParam int revisionId
+	 *
+	 * @throws MissingParameterApiException
+	 * @throws NotFoundException
+	 * @throws BadRequestApiException
+	 */
+	public function addUpvote() {
```

Comment:
I think for adding a removing votes, it would be good to have a explicit response content (even just `{"success": true}`) that makes it clear to a consumer of the API that the request succeeded, rather than just relying on the absence of an exception. Maybe in the case of adding an upvote, it might even be worth returning the inserted ID which is usable when removing the added upvote.